Automattic/jetpack
# Search: search admin page tuning

#### Changes proposed in this Pull Request:
This PR contains several fix-up for the all the previous search admin PRs #20224 
- Changed layout from flexbox to grid
- Changed gaps, fonts, colors per design review
- Changed form toggle style per the design
- Added analytics

#### Jetpack product discussion
pcNPJE-7J#comment-161

#### Does this pull request change what data or activity we track or use?
- added `jetpack_search_admin_page_view` for page view
- added `jetpack_search_instant_toggle` and `jetpack_search_module_toggle` for toggles

<s>#### Desinger's review instructions:
- Checkout the branch `git pull && git checkout update/search-admin-page-tuning`
- Build `jetpack install --all && jetpack build plugins/jetpack`
- Go to local test site with Jetpack Instant Search subscription (enter URL directly `/wp-admin/admin.php?page=jetpack-search&a8ctest`)
- Use `business-plan` in URL to force Business plan for the interface e.g. `/wp-admin/admin.php?page=jetpack-search&a8ctest&business-plan`
- Use `no-active-search-purchase` to force no active Search subscription for the interface e.g. `/wp-admin/admin.php?page=jetpack-search&a8ctest&business-plan&no-active-search-purchase`

#### Three questions need answers from designer's review
- Does 'Help your visitors find exactly what they're looking for, fast' need a period to end it? from @jeherve 
- WP footer is kept as me and @jeherve all thought WordPress branding in footer is pretty important. Do you think it's acceptable?
- Currently the mocked Search interface is by HTML+CSS, do you want to make SVGs instead which might be simpler? from @bluefuton </s>

#### Testing instructions:
- Go to a site with Jetpack Instant Search subscription (enter URL directly `/wp-admin/admin.php?page=jetpack-search&a8ctest`)
- Ensure analytics `jetpack_search_admin_page_view`, `jetpack_search_instant_toggle` and `jetpack_search_module_toggle` sent correctly. e.g. `https://pixel.wp.com/t.gif?search=false&blog_id=193141071&_en=jetpack_search_module_toggle&_ui=9025583&_ut=wpcom%3Auser_id&_ul=kangzj&_ts=1627602850078&_tz=-12&_lg=en-US&_pf=MacIntel&_ht=1120&_wd=1792&_sx=0&_sy=0&_dl=https%3A%2F%2Fjasper.au.ngrok.io%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack-search%26a8ctest%26business-plan%26no-active-search-purchase&_dr=&_rt=1627602850078&_=_`
- Ensure the page works for different screen resolutions

<s>Note:
- Discovered an issue when Search subscription is already expired, Instant Search is still in function.
- The hack to force Business plan and search subscription is only for the test/review purpose, will be reverted afterwards.</s>

![image](https://user-images.githubusercontent.com/1425433/129275954-a88a9d00-7ca4-4cf4-a7b4-69a5ac11dc93.png)
